src/Metadata/Resource/Factory/UriTemplateResourceMetadataCollectionFactory.php

```diff
@@ -139,7 +139,7 @@ private function configureUriVariables(ApiResource|HttpOperation $operation): Ap
             return $this->normalizeUriVariables($operation);
         }
 
-        $hasUserConfiguredUriVariables = !($operation->getExtraProperties['is_legacy_resource_metadata'] ?? false);
+        $hasUserConfiguredUriVariables = !($operation->getExtraProperties()['is_legacy_resource_metadata'] ?? false);
```

Comment:
@soyuka this is likely a bug in 2.7 too. And this should probably be review in 3.0.

Reply:
PR created for 2.7 branch https://github.com/api-platform/core/pull/5294